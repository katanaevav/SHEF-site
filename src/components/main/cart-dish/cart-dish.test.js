import React from "react";
import renderer from "react-test-renderer";
import CartDish from "./cart-dish.jsx";

const MockDish = {
  dishId: 1,
  dishTypeId: 1,
  dishName: `Анатомия вкуса`,
  dishDescription: `Описание блюда Описание блюда
                    Описание блюда Описание блюда
                    Описание блюда Описание блюда
                    Описание блюда Описание блюда`,
  dishPrice: `250 р.`,
  dishWeight: `250 гр.`,
  dishTag: `NEW`,
  dishImage: `./img/dish.png`,
  dishImage2x: `./img/dish@2x.png`,
  dishCount: 3,
};

it(`Render Cart Dish`, () => {
  const tree = renderer.create(
      <CartDish
        dish = {MockDish}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
