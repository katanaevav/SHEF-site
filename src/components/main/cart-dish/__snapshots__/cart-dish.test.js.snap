// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Cart Dish 1`] = `
<li
  className="shopping__item"
>
  <img
    alt="Анатомия вкуса"
    className="shopping__image"
    src="./img/dish.png"
    srcSet="./img/dish@2x.png 2x"
  />
  <div
    className="shopping__wrapper shopping__wrapper--1"
  >
    <div
      className="shopping__wrapper shopping__wrapper--2"
    >
      <p
        className="shopping__name"
      >
        Анатомия вкуса
      </p>
      <p
        className="shopping__weight"
      >
        250 гр.
      </p>
    </div>
    <div
      className="shopping__wrapper shopping__wrapper--3"
    >
      <div
        className="dish__counter select-count select-count--background-white"
      >
        <button
          className="select-count__button select-count__button--dec"
          onClick={[Function]}
        >
          -
        </button>
        <input
          className="select-count__input"
          defaultValue={3}
          max={99}
          min={1}
          onChange={[Function]}
          type="number"
        />
        <button
          className="select-count__button select-count__button--inc"
          onClick={[Function]}
        >
          +
        </button>
      </div>
      <p
        className="shopping__price"
      >
        250 р.
      </p>
    </div>
  </div>
  <a
    className="shopping__delete"
    href="#"
  >
    <img
      alt="Удалить"
      className="shopping__delete-image"
      src="./img/delete.svg"
    />
  </a>
</li>
`;
